### PR TITLE
INF-1305 export user groups json

### PR DESCRIPTION
**Pull request by nkosimarz**
Get raw groups from db for export groups (and users in groups) to json

### CODE REVIEW

File: roledb.go [hunk 1]
```diff
@@ -335,6 +342,27 @@ func (x *sqlGroupDB) GetGroups() ([]*AuthGroup, error) {
...
+
+	var groups []RawAuthGroup
+	for rows.Next() {
+		name := ""
+		permList := ""
+		var id GroupIDU32
+		if errScan := rows.Scan(&id, &name, &permList); errScan != nil {
+			return nil, errScan
+		}
+		groups = append(groups, RawAuthGroup{ID: id, Name: name, PermList: permList})
```
Comment (FritzOnFire) [inline roledb.go:361]: So for this you can just create the `RawAuthGroup` and then use it for the scan. Like
```go
r = RawAuthGroup{}
Scan(&r.id, &r.name, r.permList)
```
Makes things a bit simpler.

File: roledb.go [hunk 2]
```diff
@@ -408,6 +436,17 @@ type RoleGroupCache struct {
...
 
+// GetGroupsRaw's results are not cached. The point is to get the current state of the db for the export
+func (x *RoleGroupCache) GetGroupsRaw() ([]RawAuthGroup, error) {
+	x.groupsLock.RLock()
+	groups, err := x.backend.GetGroupsRaw()
+	if err != nil {
+		return nil, err
+	}
+	x.groupsLock.Unlock()
+	return groups, nil
```
Comment (FritzOnFire) [inline roledb.go:447]: As discussed, this should just be the one liner returning the result of the function.

**Review by FritzOnFire — CHANGES_REQUESTED** [review]
Some minor changes, but LGTM

**Review by FritzOnFire — APPROVED** [review]
LGTM

**Review by jasonkofo — APPROVED** [review]

**Review by jasonkofo — APPROVED** [review]

**Pull request merged**